servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -184,24 +212,38 @@ RouteProviders drainRoutes() {
                 final Route<Req, Resp> route, final Class<Req> requestClass,
                 final Class<Resp> responseClass, final GrpcSerializationProvider serializationProvider) {
             routes.put(path, new RouteProvider(executionContext -> toStreamingHttpService(
-                    (HttpService) (ctx, request, responseFactory) -> {
-                        try {
-                            final GrpcServiceContext serviceContext =
-                                    new DefaultGrpcServiceContext(request.path(), ctx);
-                            final HttpDeserializer<Req> deserializer =
-                                    serializationProvider.deserializerFor(readGrpcMessageEncoding(request),
-                                            requestClass);
-                            return route.handle(serviceContext, request.payloadBody(deserializer))
-                                    .map(rawResp -> newResponse(responseFactory,
-                                            ctx.executionContext().bufferAllocator())
-                                            .payloadBody(uncheckedCast(rawResp),
-                                                    serializationProvider.serializerFor(serviceContext,
-                                                            responseClass)))
-                                    .recoverWith(cause -> succeeded(newErrorResponse(responseFactory, cause,
-                                            ctx.executionContext().bufferAllocator())));
-                        } catch (Throwable t) {
-                            return succeeded(newErrorResponse(responseFactory, t,
-                                    ctx.executionContext().bufferAllocator()));
+                    new HttpService() {
+                        @Override
+                        public Single<HttpResponse> handle(final HttpServiceContext ctx, final HttpRequest request,
+                                                           final HttpResponseFactory responseFactory) {
+                            try {
+                                final GrpcServiceContext serviceContext =
+                                        new DefaultGrpcServiceContext(request.path(), ctx);
+                                final HttpDeserializer<Req> deserializer =
+                                        serializationProvider.deserializerFor(readGrpcMessageEncoding(request),
+                                                requestClass);
+                                return route.handle(serviceContext, request.payloadBody(deserializer))
+                                        .map(rawResp -> newResponse(responseFactory,
+                                                ctx.executionContext().bufferAllocator())
+                                                .payloadBody(uncheckedCast(rawResp),
```

Comment:
while you are here: `uncheckedCast` shouldn't be necessary (`Class<Resp>` should be used to give type safety) ... consider removal.